blocks/api/registration.js

```diff
@@ -104,3 +111,34 @@ export function getBlockSettings( slug ) {
 export function getBlocks() {
 	return Object.values( blocks );
 }
+
+/**
+ * Validates the block settings.
+ *
+ * @param  {Object}  settings Block settings.
+ * @return {Boolean} Whether the block settings are valid.
+ */
+export function validateBlockSettings( settings ) {
+	if ( ! settings ) {
+		console.error(
+			'Block settings must specify a save() and edit() method for each block.'
+		);
+		return false;
+	}
+
+	if ( ! settings.save || 'function' !== typeof settings.save ) {
+		console.error(
+			'Block settings must specify a save() method for each block.'
+		);
+		return false;
+	}
+
+	if ( ! settings.edit || 'function' !== typeof settings.edit ) {
+		console.error(
+			'Block settings must specify a edit() method for each block.'
+		);
+		return false;
+	}
```

Comment:
Since we're adding some validation, should we validate all the mandatory attributes we have? I'm thinking of `icon`, `title` and `category`. Maybe we should also validate that the category is a known category.